misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -88,50 +99,40 @@ internal class SqsJobConsumer @Inject internal constructor(
     handlingThreads.awaitTermination(10, TimeUnit.SECONDS)
   }
 
-  internal inner class QueueReceiver(
-    queueName: QueueName,
-    private val handler: JobHandler
-  ) {
-    private val queue = queues.getForReceiving(queueName)
-    private val shouldKeepRunning = AtomicBoolean(false)
+  internal abstract inner class QueueReceiver(queueName: QueueName) {
+    val queue = queues.getForReceiving(queueName)
+    private val shouldKeepRunning = AtomicBoolean(true)
```

Comment:
I'm pretty sure `AtomicBoolean(false)` is a bug, but we didn't notice the bug because there's another bug in `QueueReceiver::run` below that means we never return `Status.NO_SCHEDULE`. I'm fixing both.

Safe shutdown was added in https://github.com/cashapp/misk/pull/2758/